main.tf

```diff
@@ -0,0 +1,214 @@
+  count         = terraform.workspace == "prod" ? 0 : 1
+  bucket        = "ccao-glue-assets-${terraform.workspace}-us-east-1"
+  force_destroy = true
+}
+
+resource "aws_s3_bucket_public_access_block" "glue_assets" {
+  count  = terraform.workspace == "prod" ? 0 : 1
+  bucket = local.s3_bucket_glue_assets
+
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+}
+
+resource "aws_s3_bucket_versioning" "glue_assets" {
+  count  = terraform.workspace == "prod" ? 0 : 1
+  bucket = local.s3_bucket_glue_assets
+
+  versioning_configuration {
+    status = "Enabled"
+  }
+}
+
+resource "aws_s3_bucket" "data_warehouse" {
+  count         = terraform.workspace == "prod" ? 0 : 1
+  bucket        = "ccao-data-warehouse-${terraform.workspace}-us-east-1"
+  force_destroy = true
+}
+
+resource "aws_s3_bucket_public_access_block" "data_warehouse" {
+  count  = terraform.workspace == "prod" ? 0 : 1
+  bucket = local.s3_bucket_data_warehouse
+
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+}
+
+resource "aws_s3_bucket_versioning" "data_warehouse" {
+  count  = terraform.workspace == "prod" ? 0 : 1
+  bucket = local.s3_bucket_data_warehouse
+
+  versioning_configuration {
+    status = "Enabled"
+  }
+}
+
+resource "aws_s3_bucket" "athena_results" {
+  count         = terraform.workspace == "prod" ? 0 : 1
+  bucket        = "ccao-athena-results-${terraform.workspace}-us-east-1"
+  force_destroy = true
+}
+
+resource "aws_s3_bucket_public_access_block" "athena_results" {
+  count  = terraform.workspace == "prod" ? 0 : 1
+  bucket = aws_s3_bucket.athena_results[0].id
+
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+}
+
+resource "aws_s3_bucket_versioning" "athena_results" {
+  count  = terraform.workspace == "prod" ? 0 : 1
+  bucket = aws_s3_bucket.athena_results[0].id
+
+  versioning_configuration {
+    status = "Enabled"
+  }
+}
+
+resource "aws_s3_object" "sales_val_flagging" {
+  bucket = local.s3_bucket_glue_assets
+  key    = "${local.s3_prefix}/sales_val_flagging.py"
+  source = "${path.module}/glue/sales_val_flagging.py"
+  etag   = filemd5("${path.module}/glue/sales_val_flagging.py")
+}
+
+resource "aws_s3_object" "flagging_script" {
+  bucket = local.s3_bucket_glue_assets
+  key    = "${local.s3_prefix}/flagging.py"
+  source = "${path.module}/glue/flagging_script_glue/flagging.py"
+  etag   = filemd5("${path.module}/glue/flagging_script_glue/flagging.py")
+}
+
+resource "aws_s3_object" "sale_flag_metadata" {
+  count  = terraform.workspace == "prod" ? 0 : 1
+  bucket = local.s3_bucket_data_warehouse
+  key    = "sale/flag/sale_flag_schema.parquet"
+  source = "${path.module}/glue/schema/sale_flag_schema.parquet"
+  etag   = filemd5("${path.module}/glue/schema/sale_flag_schema.parquet")
+}
```

Comment:
New as of last review cycle, this bucket stores query output for the staging Athena database.